### PR TITLE
Reorganized Utils

### PR DESCRIPTION
**Pull request by tpoterba**
Significantly cleaned up code by refactoring
and deleting org.broadinstitute.hail.Utils:
1.   Rich classes are moved to utils.richUtils._,
    and implicit conversions are held in a trait
    utils.richUtils.Implicits
2.   Standalone classes in Utils have been made
    independent classes under utils
3.   Miscellaneous methods have been moved to
    utils package object

### CODE REVIEW

File: src/main/scala/org/broadinstitute/hail/utils/package.scala [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+  }
+
+  def fatal(msg: String): Nothing = {
+    throw new FatalException(msg)
+  }
+
+  def hadoopFS(filename: String, hConf: hadoop.conf.Configuration): hadoop.fs.FileSystem =
+    new hadoop.fs.Path(filename).getFileSystem(hConf)
+
+  private def hadoopCreate(filename: String, hConf: hadoop.conf.Configuration): OutputStream = {
```
Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/utils/package.scala]: I considered moving these hadoop functions out to a standalone HadoopUtils trait.  Thoughts?

**Comment by danking** [comment]
I think this is definitely an improvement.

File: src/main/scala/org/broadinstitute/hail/utils/package.scala (hunk 1 above)
Comment (danking) [inline src/main/scala/org/broadinstitute/hail/utils/package.scala]: Yeah, maybe a `class` whose instances had an `hConf`?

Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/utils/package.scala]: Could also put these on a RichHadoopConfiguration:
```
hConf.create(filename)
hConf.writeTextFile(filename) { ... }
```

that doesn't look terrible.

Comment (danking) [inline src/main/scala/org/broadinstitute/hail/utils/package.scala]: Yeah I like that.

**Comment by tpoterba** [comment]
I rebased, and made the hadoopConf changes.  Please don't let this sit for too long @cseed, each rebase makes takes a few years off my life

**Pull request merged**